client/src/components/AddProductModal.jsx

```diff
@@ -0,0 +1,151 @@
+import {
+  Button,
+  Grid,
+  makeStyles,
+  Modal,
+  Paper,
+  Slide,
+  TextField,
+  Typography,
+} from "@material-ui/core";
+import IconButton from "@material-ui/core/IconButton";
+import React, { useContext } from "react";
+import ArrowBackIcon from "@material-ui/icons/ArrowBack";
+import { userContext as context } from "../providers/UsersProvider";
+
+const useStyles = makeStyles(() => ({
+  paper: {
+    outline: "none",
+    width: "650px",
+    height: "700px",
+    position: "relative",
+    padding: "20px",
+    backgroundColor: "#f2f2f2",
+  },
+  textField: {
+    borderRadius: `4px 4px 4px 4px`,
+    outline: "none",
+    padding: "7px",
+    width: "250px",
+    backgroundColor: "white",
+  },
+  button: {
+    height: 56,
+    borderRadius: 28,
+    width: 142,
+    display: "block",
+    margin: "auto",
+    marginBottom: "20px",
+    marginTop: "20px",
+  },
+  iconButton: {
+    position: "absolute",
+    left: "20px",
+    top: "20px",
+  },
+  newItem: {
+    display: "block",
+    textAlign: "center",
+    margin: "60px 0 15px 0",
+  },
+  linkToItem: {
+    textAlign: "center",
+    marginTop: "50px",
+    fontWeight: "bold",
+  },
+  grid: { width: "fitContent", margin: "20px auto 20px auto" },
+  selectListHeader: {
+    textAlign: "center",
+    marginTop: "50px",
+    fontWeight: "bold",
+  },
+}));
+
+const AddProductModal = () => {
+  const userContext = useContext(context);
+  const classes = useStyles();
+
+  const { isAddingProd, updateIsAddingProd } = userContext;
+
+  const submitHandler = () => {
+    //Use a route for adding a product to list
+  }
+
+  return (
+    <Modal
+      open={isAddingProd}
+      onBackdropClick={() => updateIsAddingProd(false)}
+    >
```

Comment:
instead of Modal, consider using Material UI Dialog. It is a wrapper around Modal with more features. Also my thought was to not make this a separate modal, but to make it just the contents, and put it in the "EditList" modal and only slide it in if the Add Product button is clicked.

Reply:
If I'm understanding this correctly, dont make it another modal, instead make the AddProductModals content be inside of the EditList and just slide it in from the right side of the EditList Modal instead of the browser window  when the Add Product is clicked?